tests/receivers/mysql/bundled_test.go

```diff
@@ -0,0 +1,62 @@
+// Copyright Splunk, Inc.
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package tests
+
+import (
+	"fmt"
+	"path"
+	"runtime"
+	"testing"
+
+	"github.com/signalfx/splunk-otel-collector/tests/testutils"
+)
+
+func TestMysqlDockerObserver(t *testing.T) {
+	port := "3306"
+	mysql := []testutils.Container{testutils.NewContainer().WithContext(
+		path.Join(".", "testdata", "server"),
+	).WithName("mysql").WithExposedPorts(port + ":" + port).WillWaitForPorts(port).WillWaitForLogs("")}
+	testutils.SkipIfNotContainerTest(t)
+	if runtime.GOOS == "darwin" {
+		t.Skip("unable to share sockets between mac and d4m vm: https://github.com/docker/for-mac/issues/483#issuecomment-758836836")
+	}
+
+	testutils.AssertAllMetricsReceived(t, "bundled.yaml", "all_metrics_config.yaml",
```

Comment:
You are providing a config that contains a mysql receiver entry, populating it w/ the auth env vars below. The aim of discovery mode is to not require that, so please use just the otlp exporter containing one like is done in other discovery tests.